astropy/io/fits/hdu/compressed.py

```diff
@@ -677,7 +695,7 @@ def __init__(
                 header,
                 name,
                 compression_type=compression_type,
-                tile_size=tile_size,
+                tile_size=list(tile_shape[::-1]) if tile_shape else None,
```

Comment:
I haven't changed all the code in ``_update_header_data`` as this is just internal and because it is to do with headers it is actually more straightforward to deal with it in header order.